torchbenchmark/score/compute_score_v1.py

```diff
@@ -0,0 +1,193 @@
+        test, model_name, device, mode = re.match(r"test_(.*)\[(.*)\-(.*)\-(.*)\]", name).groups()
+        return (test, model_name, device, mode)
+
+    def _config_to_weight(self, config):
+        # config weight rule in V1: 1x CPU Training, 2x GPU Training, 2x CPU Inference, 2x GPU Inference
+        test, model_name, device, mode = config
+        if test == "train" and device == "cpu":
+            return 1.0
+        else:
+            return 2.0
+        
+    # Generate the domain weights from the ref object
+    def _setup_weights(self, ref):
+        # Setup domain_weights
+        domain_weights = defaultdict(float)
+        config_weights = defaultdict(float)
+        config_dict = defaultdict(lambda: defaultdict(int))
+        task_dict = defaultdict(lambda: defaultdict(set))
+        name_dict = defaultdict(lambda: defaultdict(Enum))
+        for name in ref['benchmarks']:
+            config = self._test_to_config(name)
+            task = _get_model_task(config[1])
+            # print(f"Found domain {type(task).__name__}, task: {task.name} in model {name}")
+            task_dict[type(task).__name__][task.name].add(config[1])
+            config_dict[config[1]][name] = self._config_to_weight(config)
+            name_dict[name] = task
+        category_cnt = len(task_dict)
+        sum_domain = 0.0
+        for name in name_dict:
+            task = name_dict[name]
+            domain_cnt = len(task_dict[type(task).__name__])
+            task_cnt = len(task_dict[type(task).__name__][task.name])
+            domain_weights[name] = (1.0 / category_cnt) * (1.0 / domain_cnt) * (1.0 / task_cnt)
+        # Setup config_weights
+        for name in name_dict:
+            (test, model_name, device, mode) = self._test_to_config(name)
+            config_weights[name] = config_dict[model_name][name] / sum(config_dict[model_name].values())
+        for name in name_dict:
+            total_weight = config_weights[name] * domain_weights[name]
+            sum_domain += total_weight
+        assert(abs(sum_domain - 1.0) < 1e-6), "The total task weights sum is not 1.0, please submit a bug report."
+        return (domain_weights, config_weights)
+ 
+    def _setup_benchmark_norms(self, ref_data):
+        """
+        Helper function which gets the normalization values per benchmark
+        by going through the reference data file.
+        """
+        if ref_data == TORCHBENCH_V1_REF_DATA:
+            with open(ref_data) as ref_file:
+                ref = yaml.full_load(ref_file)
+        else:
+            ref = self._get_norm_from_ref_data(ref_data)
+        return ref
+
+    def _get_norm_from_ref_data(self, ref_data):
+        """
+        This function iterates over the reference data (json object)
+        and calculates the normalization values based on the reference data.
+        It also sets up the domain weights of the score.
+        """
+        norm = {}
+        norm['benchmarks'] = {}
```

Comment:
are all the complexities of this data structure necessary?  

for instance, does `norm` need to have a top-level `benchmarks` entry,  or can it just start by indexing into names?

does `d` need to be another dict, or can it just be a scalar 'mean' value?

Reply:
The top-level benchmarks is not really needed. I will remove it later.
`d` has to be another dict here, but I can further simplify this by using `defaultdict`.